integration/test_tenants.py

```diff
@@ -0,0 +1,264 @@
+    )
+
+    collection.tenants.create([Tenant(name="Tenant1"), Tenant(name="Tenant2")])
+    tenant1 = collection.with_tenant("Tenant1")
+    tenant2 = collection.with_tenant("Tenant2")
+
+    tenant1.data.insert({"name": "some name"})
+    objects = tenant1.query.fetch_objects().objects
+    assert len(objects) == 1
+    assert objects[0].properties["name"] == "some name"
+
+    objects = tenant2.query.fetch_objects().objects
+    assert len(objects) == 0
+
+    tenant2.data.insert({"name": "some other name"})
+    objects = tenant2.query.fetch_objects().objects
+    assert len(objects) == 1
+    assert objects[0].properties["name"] == "some other name"
+
+
+def test_exist_with_tenant(collection_factory: CollectionFactory) -> None:
+    collection = collection_factory(
+        vectorizer_config=Configure.Vectorizer.none(),
+        multi_tenancy_config=Configure.multi_tenancy(enabled=True),
+    )
+    collection.tenants.create([Tenant(name="Tenant1"), Tenant(name="Tenant2")])
+
+    uuid1 = collection.with_tenant("Tenant1").data.insert({})
+    uuid2 = collection.with_tenant("Tenant2").data.insert({})
+
+    assert collection.with_tenant("Tenant1").data.exists(uuid1)
+    assert not collection.with_tenant("Tenant2").data.exists(uuid1)
+    assert collection.with_tenant("Tenant2").data.exists(uuid2)
+    assert not collection.with_tenant("Tenant1").data.exists(uuid2)
+
+
+def test_tenant_with_activity(collection_factory: CollectionFactory) -> None:
+    name = "TestTenantActivity"
+    collection = collection_factory(
+        name=name,
+        vectorizer_config=Configure.Vectorizer.none(),
+        multi_tenancy_config=Configure.multi_tenancy(enabled=True),
+    )
+    collection.tenants.create(
+        [
+            Tenant(name="1", activity_status=TenantActivityStatus.HOT),
+            Tenant(name="2", activity_status=TenantActivityStatus.COLD),
+            Tenant(name="3"),
+        ]
+    )
+    tenants = collection.tenants.get()
+    assert tenants["1"].activity_status == TenantActivityStatus.HOT
+    assert tenants["2"].activity_status == TenantActivityStatus.COLD
+    assert tenants["3"].activity_status == TenantActivityStatus.HOT
+
+
+def test_update_tenant(collection_factory: CollectionFactory) -> None:
+    name = "TestUpdateTenant"
+    collection = collection_factory(
+        name=name,
+        vectorizer_config=Configure.Vectorizer.none(),
+        multi_tenancy_config=Configure.multi_tenancy(enabled=True),
+    )
+    collection.tenants.create(Tenant(name="1", activity_status=TenantActivityStatus.HOT))
+    tenants = collection.tenants.get()
+    assert tenants["1"].activity_status == TenantActivityStatus.HOT
+
+    collection.tenants.update(Tenant(name="1", activity_status=TenantActivityStatus.COLD))
+    tenants = collection.tenants.get()
+    assert tenants["1"].activity_status == TenantActivityStatus.COLD
+
+
+def test_tenant_exists(collection_factory: CollectionFactory) -> None:
+    name = "TestTenantExists"
```

Comment:
just FYI: we do not need a name anymore as the collection name is auto-detemrined by the collection_factory by the test-name. But also doesn't hurt. Feel free to change or not.